esp/esdllib/esdl_script.cpp

```diff
@@ -1562,6 +1664,59 @@ class CEsdlTransformOperationForEach : public CEsdlTransformOperationWithChildre
     }
 };
 
+class CEsdlTransformOperationSynchronize : public CEsdlTransformOperationWithChildren
+{
+public:
+    CEsdlTransformOperationSynchronize(IXmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationWithChildren(xpp, stag, prefix, false, nullptr)
+    {
+    }
+
+    virtual ~CEsdlTransformOperationSynchronize(){}
+
+    virtual bool exec(CriticalSection *externalCrit, IInterface *stateIn, IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) override
+    {
+        if (!m_children.length())
+            return false;
+
+        IPointerArray preps;
+        ForEachItemIn(i, m_children)
+            preps.append(m_children.item(i).prepare(scriptContext, targetContext, sourceContext));
+
+        class casyncfor: public CAsyncFor
+        {
+            CriticalSection synchronizeCrit;
+            IPointerArray &preps;
+            IArrayOf<IEsdlTransformOperation> &children;
+            IEsdlScriptContext *scriptContext = nullptr;
+            IXpathContext *targetContext = nullptr;
+            IXpathContext *sourceContext = nullptr;
+
+        public:
+            casyncfor(IPointerArray &_preps, IArrayOf<IEsdlTransformOperation> &_children, IEsdlScriptContext *_scriptContext, IXpathContext *_targetContext, IXpathContext *_sourceContext)
+                : preps(_preps), children(_children), scriptContext(_scriptContext), targetContext(_targetContext), sourceContext(_sourceContext)
+            {
+            }
+            void Do(unsigned i)
+            {
+                children.item(i).exec(&synchronizeCrit, preps.item(i), scriptContext, targetContext, sourceContext);
+            }
+        } afor(preps, m_children, scriptContext, targetContext, sourceContext);
+
+        afor.For(m_children.ordinality(), 10, false, false);
```

Comment:
max at once - 10 - should probably be configurable.

Reply:
done.